src/components/inputBar.tsx

```diff
@@ -418,20 +437,41 @@ const InputBar = (props: InputBarProps) => {
               animate={{ x: 0, opacity: 1, transition: { duration: 0.5 } }}
               exit={{ x: 50, opacity: 0, transition: { duration: 0.5 } }}
             >
-              <Button
-                size="icon"
-                variant="secondary"
-                disabled={
-                  props.isChatCompleted ||
-                  isRecording ||
-                  isTranscribing ||
-                  disableInputs
-                }
-                type="submit"
-                className="disabled:text-muted"
-              >
-                <PaperPlaneTilt className="h-4 w-4 fill-current" />
-              </Button>
+              {props.dropZone ? (
+                <>
+                  <Button
+                    size="icon"
+                    variant="secondary"
+                    disabled={
+                      props.isChatCompleted ||
+                      isRecording ||
+                      isTranscribing ||
+                      disableInputs
+                    }
+                    onClick={handleDropzone}
```

Comment:
Instead of calling this callback here, call it inside the handleSubmit and it won't require you to have an explicit state for imageInput

Reply:
resolved it